os/osx/window.mm

```diff
@@ -120,6 +120,8 @@ - (WindowOSXObjc*)initWithImpl:(os::WindowOSX*)impl
     if (spec->floating()) {
       self.level = NSFloatingWindowLevel;
       self.hidesOnDeactivate = true;
+      if (spec->modal())
+        [self setLevel:NSModalPanelWindowLevel];
     }
```

Comment:
We're already setting the level property if it's floating, probably we can have a modal / not floating (this is not the case of Aseprite which sets the "floating" property for all non-main windows):
```suggestion
    if (spec->floating()) {
      self.level = NSFloatingWindowLevel;
      self.hidesOnDeactivate = true;
    }
    if (spec->modal())
      self.level = NSModalPanelWindowLevel;
```